Tests broken on Python 3.8 since #507

#507 broke the test suite on Python 3.8 (not on higher versions of Python)

https://github.com/capytaine/capytaine/actions/runs/8964962350

```
  ___________ test_symmetry_of_the_green_function_infinite_depth[gf1] ____________
  
  gf = XieDelhommeau(tabulation_nr=328, tabulation_rmax=100.0, tabulation_nz=46, tabulation_zmin=-251.0, tabulation_nb_integr...y', finite_depth_prony_decomposition_method='fortran', floating_point_precision='float64', gf_singularities='low_freq')
  
      @pytest.mark.parametrize("gf", gfs)
      def test_symmetry_of_the_green_function_infinite_depth(gf):
          k = 1.0
          xi = np.array([0.0, 0.0, -1.0])
          xj = np.array([1.0, 1.0, -2.0])
  >       g1, dg1 = gf.fortran_core.green_wave.wave_part_infinite_depth(
              xi, xj, k,
              gf.tabulation_nb_integration_points, gf.tabulation_grid_shape_index,
              gf.tabulated_r_range, gf.tabulated_z_range,
              gf.tabulated_integrals, gf.gf_singularities_index
          )
  E       TypeError: Delhommeau_float64.green_wave.wave_part_infinite_depth() takes at most 0 arguments (9 given)
```

The simplest option is to already drop support for 3.8 (official end of life in October 2024).

--- a/capytaine/green_functions/libDelhommeau/src/Green_wave.f90
+++ b/capytaine/green_functions/libDelhommeau/src/Green_wave.f90
@@ -161,13 +161,13 @@ CONTAINS
   ! =====================================================================
 
   SUBROUTINE WAVE_PART_INFINITE_DEPTH                            &
-      ! Returns (G^-, nabla G^+) if gf_singularities == HIGH_FREQ
-      ! and (G^+, nabla G^+) if gf_singularities == LOW_FREQ
       (X0I, X0J, wavenumber,                                     &
       tabulation_nb_integration_points, tabulation_grid_shape,   &
       tabulated_r_range, tabulated_z_range, tabulated_integrals, &
       gf_singularities,                                          &
       G, nablaG)
+      ! Returns (G^-, nabla G^+) if gf_singularities == HIGH_FREQ
+      ! and (G^+, nabla G^+) if gf_singularities == LOW_FREQ
 
     ! Inputs
     REAL(KIND=PRE), DIMENSION(3),             INTENT(IN) :: X0I, X0J

